Content.Goobstation.Server/Xenobiology/SlimeGrinder/SlimeGrinderSystem.cs

```diff
@@ -0,0 +1,184 @@
+        while (query.MoveNext(out var uid, out _, out var grinder))
+        {
+            grinder.ProcessingTimer -= frameTime;
+
+            if (grinder.ProcessingTimer > 0
+                || !TryComp<SlimeComponent>(grinder.EntityGrinded, out var slime))
+                continue;
+
+            var extractProto = _xenobio.GetProducedExtract((grinder.EntityGrinded.Value, slime));
+            SpawnNextToOrDrop(extractProto, uid);
+
+            QueueDel(grinder.EntityGrinded);
+            grinder.EntityGrinded = null;
+
+            RemCompDeferred<ActiveSlimeGrinderComponent>(uid);
+        }
+    }
+
+    #region  Active Grinding
+
+    private void OnActiveInit(Entity<ActiveSlimeGrinderComponent> activeGrinder, ref ComponentInit args)
+    {
+        if (!TryComp<SlimeGrinderComponent>(activeGrinder, out var grinder))
+            return;
+
+        _jitteringSystem.AddJitter(activeGrinder, -10, 100);
+        _sharedAudioSystem.PlayPvs(grinder.GrindSound, activeGrinder);
+        _ambientSoundSystem.SetAmbience(activeGrinder, true);
+    }
+
+    private void OnActiveShutdown(Entity<ActiveSlimeGrinderComponent> activeGrinder, ref ComponentShutdown args)
+    {
+        RemComp<JitteringComponent>(activeGrinder);
+        _ambientSoundSystem.SetAmbience(activeGrinder, false);
+    }
+
+    private void OnUnanchorAttempt(Entity<ActiveSlimeGrinderComponent> activeGrinder, ref UnanchorAttemptEvent args) =>
+        args.Cancel();
+
+    private void OnPowerChanged(EntityUid uid, SlimeGrinderComponent component, ref PowerChangedEvent args)
+    {
+        if (args.Powered)
+        {
+            if (component.ProcessingTimer > 0)
+                EnsureComp<ActiveSlimeGrinderComponent>(uid);
+        }
+        else
+            RemComp<ActiveSlimeGrinderComponent>(uid);
+    }
+
+    #endregion
+
+    private void OnInit(Entity<SlimeGrinderComponent> grinder, ref MapInitEvent args) =>
+        grinder.Comp.GrindedContainer = _container.EnsureContainer<Container>(grinder, "GrindedContainer");
+
+    private void OnShutdown(Entity<SlimeGrinderComponent> grinder, ref ComponentShutdown args) =>
+        _container.EmptyContainer(grinder.Comp.GrindedContainer);
+    private void OnUnanchored(Entity<SlimeGrinderComponent> grinder, ref BeforeUnanchoredEvent args) =>
+        _container.EmptyContainer(grinder.Comp.GrindedContainer);
+    private void OnAfterInteractUsing(Entity<SlimeGrinderComponent> grinder, ref AfterInteractUsingEvent args)
+    {
+        if (!args.CanReach
+            || args.Target == null
+            || !TryComp<PhysicsComponent>(args.Used, out var physics)
+            || !CanGrind(grinder, args.Used))
+            return;
+
+        var delay = grinder.Comp.BaseInsertionDelay * physics.FixturesMass;
+        _doAfter.TryStartDoAfter(new DoAfterArgs(EntityManager, args.User, delay, new ReclaimerDoAfterEvent(), grinder, target: args.Target, used: args.Used)
```

Comment:
Break up large lines like that. Annoying to read on smaller displays (such as mr steamdeck's) Also i think some of these could be moved inside the bracket? dont remember
```suggestion
        _doAfter.TryStartDoAfter(new DoAfterArgs(EntityManager,
            args.User,
            delay,
            new ReclaimerDoAfterEvent(),
            grinder,
            target: args.Target,
            used: args.Used)